main.go

```diff
@@ -81,32 +85,74 @@ func printVersion() {
 	fmt.Println(version)
 }
 
-func generateFromRequest(pkgDir, structModelName, repositoryInterfaceName string) (string, error) {
-	fset := token.NewFileSet()
-	dir, err := parser.ParseDir(fset, pkgDir, nil, parser.ParseComments)
+var (
+	errStructNotFound    = errors.New("struct not found")
+	errInterfaceNotFound = errors.New("interface not found")
+)
+
+func generateFromRequest(
+	pkgDir, modelDirPtr, structModelName, repositoryInterfaceName, destPkgName string,
+) (string, error) {
+	pkg, err := parsePkg(pkgDir)
 	if err != nil {
 		return "", err
 	}
+	if destPkgName == "" {
+		destPkgName = pkg.Name
+	}
+	if pkgDir == modelDirPtr {
+		structModel, ok := pkg.Structs[structModelName]
+		if !ok {
+			return "", errStructNotFound
+		}
+		return generateRepository(pkg, structModel, repositoryInterfaceName, "", destPkgName)
+	} else {
+		modelPkg, err := parsePkg(modelDirPtr)
+		if err != nil {
+			return "", err
+		}
+		structModel, ok := modelPkg.Structs[structModelName]
+		if !ok {
+			return "", errStructNotFound
+		}
+		structModel.PackageAlias = modelPkg.Name
+		return generateRepository(pkg, structModel, repositoryInterfaceName, modelPkg.Path, destPkgName)
+	}
+}
 
-	pkg, err := code.ParsePackage(dir)
-	if err != nil {
-		return "", err
+func parsePkg(pkgDir string) (code.Package, error) {
+	dirParser := func(dir string) (pkgs map[string]*ast.Package, err error) {
+		return parser.ParseDir(token.NewFileSet(), dir, nil, parser.ParseComments)
 	}
 
-	return generateRepository(pkg, structModelName, repositoryInterfaceName)
+	pkgPserser := code.NewPackageParser(dirParser, parsePackageID)
```

Comment:
1. Typo for variable name.
2. `dirParser` and `parsePackageID` should be inside `internal/code` directory.
3. I don't really like passing function through function parameters. I understand that it helps for testing, but I am still thinking about the solution. While the solution is not ready, you may set your `dirParser` and `parsePackageID` as a default and make it overridable in tests.

Reply:
thanks for suggestion!

fixed in 6089dff847d46eb528584d4c300990a095cfd89d

since test file is `code_test`, it cannot access private field of `PackageParser`. Therefore I have to make it public.

Reply:
That's ok. I will look into it later.